gateway/src/shard/builder.rs

```diff
@@ -200,35 +200,21 @@ impl ShardBuilder {
     }
 
     pub(crate) fn into_config(self) -> Config {
-        let Self {
-            event_types,
-            gateway_url,
-            http_client,
-            identify_properties,
-            intents,
-            large_threshold,
-            presence,
-            queue,
-            shard,
-            token,
-            ratelimit_payloads,
-        } = self;
```

Comment:
out of curiosity, what's the reason not to destructure self here?

Reply:
It just introduced complexity, specifically in #1601 where not every field needed to be destructured.